<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>676</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Select Features</string>
  </property>
  <!-- Top-level vertical layout to allow resizing -->
  <layout class="QVBoxLayout" name="verticalLayout_Main">
    <!-- Define ROI Section -->
    <item>
      <widget class="QGroupBox" name="groupBox_17">
        <property name="title">
          <string>Define ROI</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_GroupBox17">
          <item row="0" column="0">
            <widget class="QLabel" name="label_roiName_4">
              <property name="text">
                <string>Name</string>
              </property>
              <property name="font">
                <font>
                  <family>MS Shell Dlg 2</family>
                  <pointsize>8</pointsize>
                </font>
              </property>
            </widget>
          </item>
          <item row="0" column="1">
            <widget class="QLabel" name="label_6">
              <property name="text">
                <string>Color Clusters</string>
              </property>
              <property name="font">
                <font>
                  <family>MS Shell Dlg 2</family>
                  <pointsize>8</pointsize>
                </font>
              </property>
            </widget>
          </item>
          <item row="1" column="0">
            <widget class="QLineEdit" name="lineEdit_roiName">
              <property name="font">
                <font>
                  <family>MS Shell Dlg 2</family>
                  <pointsize>8</pointsize>
                </font>
              </property>
            </widget>
          </item>
          <item row="1" column="1">
            <widget class="QSpinBox" name="spinBoxColorClusters">
              <property name="minimum">
                <number>1</number>
              </property>
              <property name="maximum">
                <number>8</number>
              </property>
              <property name="value">
                <number>4</number>
              </property>
              <property name="font">
                <font>
                  <family>MS Shell Dlg 2</family>
                  <pointsize>8</pointsize>
                </font>
              </property>
            </widget>
          </item>
          <item row="1" column="2">
            <widget class="QPushButton" name="pushButtonAddROI">
              <property name="text">
                <string>Add</string>
              </property>
              <property name="default">
                <bool>true</bool>
              </property>
              <property name="font">
                <font>
                  <family>MS Shell Dlg 2</family>
                  <pointsize>8</pointsize>
                  <weight>75</weight>
                  <bold>true</bold>
                </font>
              </property>
            </widget>
          </item>
          <!-- Embedded Shape Group Box -->
          <item row="2" column="0" colspan="3">
            <widget class="QGroupBox" name="groupBox_18">
              <property name="title">
                <string>Shape</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_Shape">
                <item>
                  <widget class="QRadioButton" name="radioButton_ROIShapeRectangle">
                    <property name="text">
                      <string>Rectangle</string>
                    </property>
                    <property name="font">
                      <font>
                        <family>MS Shell Dlg 2</family>
                        <pointsize>8</pointsize>
                      </font>
                    </property>
                  </widget>
                </item>
                <item>
                  <spacer name="horizontalSpacer_8">
                    <property name="orientation">
                      <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                      <size>
                        <width>98</width>
                        <height>13</height>
                      </size>
                    </property>
                  </spacer>
                </item>
                <item>
                  <widget class="QRadioButton" name="radioButton_ROIShapeEllipse">
                    <property name="text">
                      <string>Ellipse</string>
                    </property>
                    <property name="font">
                      <font>
                        <family>MS Shell Dlg 2</family>
                        <pointsize>8</pointsize>
                      </font>
                    </property>
                  </widget>
                </item>
              </layout>
            </widget>
          </item>
          <!-- Delete All ROIs Button -->
          <item row="3" column="0" colspan="3">
            <widget class="QPushButton" name="pushButton_deleteAllROIs">
              <property name="text">
                <string>Delete All ROIs</string>
              </property>
              <property name="font">
                <font>
                  <family>MS Shell Dlg 2</family>
                  <pointsize>8</pointsize>
                </font>
              </property>
            </widget>
          </item>
          <!-- Layout for additional checkboxes -->
          <item row="4" column="0" colspan="3">
            <widget class="QWidget" name="layoutWidget_4">
              <layout class="QGridLayout" name="gridLayout_4">
                <item row="0" column="0">
                  <widget class="QCheckBox" name="checkBoxDisplayROIs">
                    <property name="text">
                      <string>Display ROIs</string>
                    </property>
                    <property name="checked">
                      <bool>true</bool>
                    </property>
                    <property name="font">
                      <font>
                        <family>MS Shell Dlg 2</family>
                        <pointsize>8</pointsize>
                      </font>
                    </property>
                  </widget>
                </item>
                <item row="0" column="2">
                  <widget class="QCheckBox" name="checkBoxCalcColors">
                    <property name="text">
                      <string>Display ROI  Colors</string>
                    </property>
                    <property name="checked">
                      <bool>true</bool>
                    </property>
                    <property name="toolTip">
                      <string>Check this box to only train on colors when 'Add' is clicked.</string>
                    </property>
                    <property name="font">
                      <font>
                        <family>MS Shell Dlg 2</family>
                        <pointsize>8</pointsize>
                      </font>
                    </property>
                  </widget>
                </item>
              </layout>
            </widget>
          </item>
        </layout>
      </widget>
    </item>
    <!-- Feature Calculations Section -->
    <item>
      <widget class="QGroupBox" name="groupBox_ScalarOutputs">
        <property name="title">
          <string>Feature Calculations</string>
        </property>
        <property name="flat">
          <bool>true</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_ScalarOutputs">
          <!-- Region Select Subsection -->
          <item>
            <widget class="QGroupBox" name="groupBox_14">
              <property name="title">
                <string>Region Select</string>
              </property>
              <property name="toolTip">
                <string>Choose whether to save stats for the ROI, for the Image, or both.</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_RegionSelect">
                <item>
                  <widget class="QCheckBox" name="checkBoxScalarRegion_WholeImage">
                    <property name="text">
                      <string>Whole Image</string>
                    </property>
                    <property name="checked">
                      <bool>false</bool>
                    </property>
                    <property name="font">
                      <font>
                        <family>MS Shell Dlg 2</family>
                        <pointsize>8</pointsize>
                      </font>
                    </property>
                  </widget>
                </item>
                <item>
                  <widget class="QCheckBox" name="checkBoxScalarRegion_ROI">
                    <property name="text">
                      <string>ROI</string>
                    </property>
                    <property name="checked">
                      <bool>true</bool>
                    </property>
                    <property name="font">
                      <font>
                        <family>MS Shell Dlg 2</family>
                        <pointsize>8</pointsize>
                      </font>
                    </property>
                  </widget>
                </item>
              </layout>
            </widget>
          </item>
          <!-- Feature Select Subsection -->
          <item>
            <widget class="QGroupBox" name="groupBox_15">
              <property name="title">
                <string>Feature Select</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_FeatureSelect">
                <item>
                  <widget class="QCheckBox" name="checkBox_ShannonEntropy">
                    <property name="text">
                      <string>Shannon Entropy</string>
                    </property>
                    <property name="checked">
                      <bool>true</bool>
                    </property>
                    <property name="font">
                      <font>
                        <family>MS Shell Dlg 2</family>
                        <pointsize>8</pointsize>
                      </font>
                    </property>
                  </widget>
                </item>
                <item>
                  <widget class="QCheckBox" name="checkBox_Intensity">
                    <property name="text">
                      <string>Intensity</string>
                    </property>
                    <property name="checked">
                      <bool>true</bool>
                    </property>
                    <property name="font">
                      <font>
                        <family>MS Shell Dlg 2</family>
                        <pointsize>8</pointsize>
                      </font>
                    </property>
                  </widget>
                </item>
                <item>
                  <widget class="QCheckBox" name="checkBoxColor_HSV">
                    <property name="text">
                      <string>Color (HSV)</string>
                    </property>
                    <property name="checked">
                      <bool>true</bool>
                    </property>
                    <property name="font">
                      <font>
                        <family>MS Shell Dlg 2</family>
                        <pointsize>8</pointsize>
                      </font>
                    </property>
                  </widget>
                </item>
                <item>
                  <widget class="QCheckBox" name="checkBox_Texture">
                    <property name="text">
                      <string>Texture</string>
                    </property>
                    <property name="font">
                      <font>
                        <family>MS Shell Dlg 2</family>
                        <pointsize>8</pointsize>
                      </font>
                    </property>
                  </widget>
                </item>
                <!-- Greenness Subgroup -->
                <item>
                  <widget class="QGroupBox" name="groupBox_16">
                    <property name="title">
                      <string>Greenness</string>
                    </property>
                    <layout class="QVBoxLayout" name="verticalLayout_Greenness">
                      <item>
                        <widget class="QCheckBox" name="checkBox_GCC">
                          <property name="text">
                            <string>GCC (Greenness Index)</string>
                          </property>
                          <property name="checked">
                            <bool>true</bool>
                          </property>
                          <property name="font">
                            <font>
                              <family>MS Shell Dlg 2</family>
                              <pointsize>8</pointsize>
                            </font>
                          </property>
                        </widget>
                      </item>
                      <item>
                        <widget class="QCheckBox" name="checkBox_GLI">
                          <property name="text">
                            <string>GLI (Green Leaf Index)</string>
                          </property>
                          <property name="font">
                            <font>
                              <family>MS Shell Dlg 2</family>
                              <pointsize>8</pointsize>
                            </font>
                          </property>
                        </widget>
                      </item>
                      <item>
                        <widget class="QCheckBox" name="checkBox_ExG">
                          <property name="text">
                            <string>ExG (Excess Green Index)</string>
                          </property>
                          <property name="font">
                            <font>
                              <family>MS Shell Dlg 2</family>
                              <pointsize>8</pointsize>
                            </font>
                          </property>
                        </widget>
                      </item>
                      <item>
                        <widget class="QCheckBox" name="checkBox_RGI">
                          <property name="text">
                            <string>RGI (Red-Green Index)</string>
                          </property>
                          <property name="font">
                            <font>
                              <family>MS Shell Dlg 2</family>
                              <pointsize>8</pointsize>
                            </font>
                          </property>
                        </widget>
                      </item>
                      <item>
                        <widget class="QCheckBox" name="checkBox_NDVI">
                          <property name="text">
                            <string>False NDVI (Norm. Dif. Veg. Index using RGB)</string>
                          </property>
                          <property name="font">
                            <font>
                              <family>MS Shell Dlg 2</family>
                              <pointsize>8</pointsize>
                            </font>
                          </property>
                        </widget>
                      </item>
                    </layout>
                  </widget>
                </item>
              </layout>
            </widget>
          </item>
        </layout>
      </widget>
    </item>
    <!-- Bottom buttons layout -->
    <item>
      <layout class="QHBoxLayout" name="horizontalLayout_Buttons">
        <item>
          <widget class="QPushButton" name="pushButton_Dlg_BuildFeatureFile">
            <property name="text">
              <string>Build
Feature File</string>
            </property>
            <property name="autoFillBackground">
              <bool>true</bool>
            </property>
            <property name="font">
              <font>
                <family>Calibri</family>
                <pointsize>10</pointsize>
                <weight>75</weight>
                <bold>true</bold>
              </font>
            </property>
          </widget>
        </item>
        <item>
          <widget class="QPushButton" name="pushButton_Dlg_TEST">
            <property name="text">
              <string>TEST</string>
            </property>
            <property name="font">
              <font>
                <family>Calibri</family>
                <pointsize>10</pointsize>
                <weight>75</weight>
                <bold>true</bold>
              </font>
            </property>
          </widget>
        </item>
        <item>
          <widget class="QDialogButtonBox" name="buttonBox_Close">
            <property name="standardButtons">
              <set>QDialogButtonBox::NoButton</set>
            </property>
            <property name="enabled">
              <bool>false</bool>
            </property>
            <property name="font">
              <font>
                <family>MS Shell Dlg 2</family>
                <pointsize>8</pointsize>
              </font>
            </property>
          </widget>
        </item>
      </layout>
    </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox_Close</sender>
   <signal>accepted()</signal>
   <receiver>Dialog</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>buttonBox_Close</sender>
   <signal>rejected()</signal>
   <receiver>Dialog</receiver>
   <slot>reject()</slot>
  </connection>
 </connections>
</ui>
